kubernetes/apps_kubeclient.go

```diff
@@ -377,40 +463,52 @@ func (kc *kubeClient) GetEnvironments() ([]*app.SimpleEnvironment, error) {
 func (kc *kubeClient) GetEnvironment(envName string) (*app.SimpleEnvironment, error) {
 	envNS, err := kc.getEnvironmentNamespace(envName)
 	if err != nil {
-		return nil, err
+		return nil, errs.WithStack(err)
 	}
 
 	envStats, err := kc.getResourceQuota(envNS)
 	if err != nil {
-		return nil, err
+		return nil, errs.WithStack(err)
 	}
 
 	env := &app.SimpleEnvironment{
-		Name:  &envName,
-		Quota: envStats,
+		Type: "environment",
+		Attributes: &app.SimpleEnvironmentAttributes{
+			Name:  &envName, // TODO UUID
+			Quota: envStats,
+		},
 	}
 	return env, nil
 }
 
 func getMetricsURLFromAPIURL(apiURLStr string) (string, error) {
+	metricsURL, err := modifyURL(apiURLStr, "metrics", "")
+	if err != nil {
+		return "", err
+	}
+	return metricsURL.String(), nil
+}
+
+func modifyURL(apiURLStr string, prefix string, path string) (*url.URL, error) {
```

Comment:
do you have a test for this function ?

Reply:
I have someone on my team writing tests, and I will get him to add coverage for this.
I would like to do that as a separate testing PR - this PR is focused on JSON-API compatiblity.